File: tools/project-memory-context/tests/fixtures/sample.rb
module Greetable
  def greet
    "hello"
  end
end

class Animal
  include Greetable

  def initialize(name)
    @name = name
  end

  def speak
    "..."
  end
end

class Dog < Animal
  def speak
    "woof"
  end
end

File: tools/project-memory-context/tests/fixtures/sample.php
<?php

interface Greetable {
    public function greet(): string;
}

abstract class Animal {
    abstract public function speak(): string;
    public function breathe(): void {}
}

class Dog extends Animal {
    public function speak(): string {
        return "woof";
    }
}

trait Loggable {
    public function log(string $msg): void {}
}

function standaloneFunction(int $x, int $y): int {
    return $x + $y;
}
